sources/tools/Stride.Importer.Assimp/MeshConverter.cs

```diff
@@ -0,0 +1,1375 @@
+            }
+        }
+
+        private bool IsNotBlackColor(System.Numerics.Vector4 diffColor)
+        {
+            return diffColor != System.Numerics.Vector4.Zero;
+        }
+
+        private unsafe void BuildLayeredSurface(Silk.NET.Assimp.Material* pMat, bool hasBaseColor, bool hasBaseValue, Color4 baseColor, float baseValue, TextureType textureType, MaterialAsset finalMaterial)
+        {
+            var nbTextures = assimp.GetMaterialTextureCount(pMat, textureType);
+
+            IComputeColor computeColorNode = null;
+            int textureCount = 0;
+            if (nbTextures == 0)
+            {
+                if (hasBaseColor)
+                {
+                    computeColorNode = new ComputeColor(baseColor);
+                }
+                //else if (hasBaseValue)
+                //{
+                //	computeColorNode = gcnew MaterialFloatComputeNode(baseValue);
+                //}
+            }
+            else
+            {
+                computeColorNode = GenerateOneTextureTypeLayers(pMat, textureType, textureCount, finalMaterial);
+            }
+
+            if (computeColorNode == null)
+            {
+                return;
+            }
+
+            if (textureType == TextureType.TextureTypeDiffuse)
+            {
+                if (assimp.GetMaterialTextureCount(pMat, TextureType.TextureTypeLightmap) > 0)
+                {
+                    var lightMap = GenerateOneTextureTypeLayers(pMat, TextureType.TextureTypeLightmap, textureCount, finalMaterial);
+                    if (lightMap != null)
+                        computeColorNode = new ComputeBinaryColor(computeColorNode, lightMap, BinaryOperator.Add);
+                }
+
+                finalMaterial.Attributes.Diffuse = new MaterialDiffuseMapFeature(computeColorNode);
+
+                // TODO TEMP: Set a default diffuse model
+                finalMaterial.Attributes.DiffuseModel = new MaterialDiffuseLambertModelFeature();
+            }
+            else if (textureType == TextureType.TextureTypeSpecular)
+            {
+                var specularFeature = new MaterialSpecularMapFeature
+                {
+                    SpecularMap = computeColorNode
+                };
+                finalMaterial.Attributes.Specular = specularFeature;
+
+                // TODO TEMP: Set a default specular model
+                var specularModel = new MaterialSpecularMicrofacetModelFeature
+                {
+                    Fresnel = new MaterialSpecularMicrofacetFresnelSchlick(),
+                    Visibility = new MaterialSpecularMicrofacetVisibilityImplicit(),
+                    NormalDistribution = new MaterialSpecularMicrofacetNormalDistributionBlinnPhong()
+                };
+                finalMaterial.Attributes.SpecularModel = specularModel;
+            }
+            else if (textureType == TextureType.TextureTypeEmissive)
+            {
+                // TODO: Add support
```

Comment:
Would be nice to support all these texture map types, as they are pretty much standard today for PBR workflows.

Reply:
Yes, fully agree, importing textures with the model would be great.

Reply:
It looks like there is already an issue for this: https://github.com/stride3d/stride/issues/473